### ISSUE TITLE
no errors but styles are not being output

### ISSUE THREAD

**Issue by wallaceturner** [kristerkari/react-native-sass-transformer#32]
I'm using expo 3.4.1 with a blank app and have followed your [setup instructions](https://github.com/kristerkari/react-native-sass-transformer) however no styles are transformed

```
import styles from "./App.scss";

console.log('styles', styles)
```

the output of this is 
`styles /static/media/App.f330be79.scss`

I'm assuming that it should actually output the `styles` object. I'm using a typescript project

**Comment by kristerkari** [comment]
Thanks! Any possibility for you to upload your project somewhere where I can have a look at it? I can't really help with the setup with only the details that you provided.

**Comment by wallaceturner** [comment]
Recreate
1) `expo init`
2) Select 'Typescript Minimal`
3) `yarn add --dev react-native-sass-transformer node-sass`
4) Create file `App.scss` and paste in css from your instructions
5) Modify `App.tsx`
   a) comment out `const styles` block
   b) add `import styles from "./App.scss";` to `App.tsx`
   c) add `console.log('styles', styles)`
6) add `metro.config.js` and edit `app.json` as per your instructions
7) yarn start


I have created a repo here: https://github.com/wallaceturner/sass-spike

**Comment by kristerkari** [comment]
sure, just a moment

**Comment by wallaceturner** [comment]
Yes, sorry I omitted the crucial instruction apologies! Could you please detail the extra setup?

**Comment by kristerkari** [comment]
Unfortunately the Web support for Expo is quite a new thing, and I haven't had the time to look at it, so I can not currently give you any instructions on how to do it.

What it would need is a small plugin (for Webpack I think) that does the same thing as the transformer (calls Sass and then parses the CSS into a style object).

**Comment by riccardolardi** [comment]
Any updates on this? I'm also not getting an object from `styles` but rather the filename.

**Comment by kristerkari** [comment]
@riccardolardi not yet, I haven't had a look at the Expo Web support yet. I'm guessing that it would need some kind of a plugin/webpack setup to get it to work.

**Comment by nikshirobokov** [comment]
> ...also not getting an object from styles but rather the filename.

Is there any progress in solving this issue? Or there may be recommendations for extra configuration for the Expo Web. 3 years after the opening of this, I'm getting a similar issue.

**Comment by nikshirobokov** [comment]
**I catch this!**

To launch the browser version, Expo Web uses webpack by default and to launch Android/iOS versions, Expo uses metro (if you doesn't change it youself).

This problem with importing styles occurs due to the fact that we configure the metro configuration and expect to get the result for the web version too, but this does not happen, since webpack is used by default for browsers. All you need to do is switch the web handler in app.json.
```
/* app.json */
{
  "expo": {
    ...,
    "packagerOpts": {
      "config": "metro.config.js",
      "sourceExts": ["ts", "tsx", "scss"] /* My project uses typescript and SCSS */
    },
    "web": {
      "bundler": "metro", /* Expo Web uses webpack by default */
      ...
    },
    ...
  }
}
```
In general, this is logical and you can read more about the difference between the build of the web version on the official Expo website: https://docs.expo.dev/guides/customizing-metro/#expo-webpack-vs-expo-metro

**Comment by DeveloperBlue** [comment]
Unfortunately for me, I cannot use the Metro bundler for web at this time. **Is there any way to get this plugin to work with Webpack?** I am on Expo 49.

One of my build targets is Electron, and the tooling for it (Electron-Forge) uses webpack to build its renderer process.
